### DISCUSSION TITLE
Module federation

### DISCUSSION THREAD

**Discussion by roshan2921** [originjs/vite-plugin-federation#172] [Q&A] [answered]
Trying to achieve module federation using vite plugin originjs vite-plugin-federation
Remote entry.js is not working after exposing to load that in host application
Any resource or blog to do module federation using this plug-in. I'm using
Vue+vite with this plug-in

**Comment by roshan2921** [comment, marked as answer]
To check the remoteEntry.js we have to build it. In dev mode we can't see that file After building the application using 
npm run build
And then in preview 
command to check preview 
npm run preview
http://localhost:4173/assets/remoteEntry.js  here you can check the remoteEntry.js working 

But how to import the remote application in shell or Host application can anyone help me with this

**Comment by Jiannan-dev** [comment]
Could you be more specific? I'm not quite sure what your problem is

**Comment by roshan2921** [comment]
Hey thank you for your reply i was facing problem while exposing
remoteEntry.js
When trying to see in
http://localhost:3000/remote entry.js here I'm not able see remoteEntry.js
But later from the discussions in GitHub i got to know that it should be
like this
http://localhost:3000/assets/remoteEntry.js

And

While importing in host application also we should import using
defineasyncComponent
And how to load the remote on some event like button click with loading it
upfront by starting the host application


My problem is solved but what i felt it is very difficult for a new person
to understand module federation by using GitHub repo
So can u please take sometime and write a simple blog how module federation
works in vue+vite using vite plugin for module federation originjs
How to expose in remote and import in Host simple guide 😀.

**Reply by Jiannan-dev** [reply]
I know what you mean, thanks for the suggestion, I remember a previous blog to describe federation, it should be this https://medium.com/@vueshenzhen/a-brief-introduction-of-vite-plugin-federation-9e998b2e43b

**Reply by Jiannan-dev** [reply]
This is a previous article, the content should not be the latest, if there is time I will try to write a blog again, but recently has been relatively busy

**Reply by roshan2921** [reply]
Thank you 👍👍

**Reply by xparagon** [reply]
But how to run a micro-frontend in dev mode? when /remoteEntry.js is not available in dev, nor is /assets/remoteEntry.js - so how to import remotes when nothing is exposed?

**Reply by itschip** [reply]
Can't you just solve it by running build --watch --write?
